src/PrestaShopBundle/Resources/config/services/adapter/product.yml

```diff
@@ -382,7 +375,8 @@ services:
 
   PrestaShop\PrestaShop\Adapter\Product\Update\ProductCategoryUpdater:
     arguments:
-      - '@PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository'
+      - '@PrestaShop\PrestaShop\Adapter\Product\Repository\ProductMultiShopRepository'
+      - '@PrestaShop\PrestaShop\Adapter\Category\Repository\CategoryRepository'
```

Comment:
autowire here + public false (same for PrestaShop\PrestaShop\Adapter\Product\Update\ProductShopUpdater)